src/lib/server/cart.ts

```diff
@@ -114,7 +116,35 @@ export async function addToCartInDb(
 	) {
 		throw error(400, 'Cart has too many items');
 	}
-
+	let variationPriceArray: Price[] = [];
+	if (params.chosenVariations) {
+		const variationNamesInDB =
+			Object.keys(product.variationLabels?.names || []).map((key) => key) || [];
+
+		const chosenVariationNames = Object.keys(params.chosenVariations);
+
+		const allVariationsChosen =
+			variationNamesInDB.length === chosenVariationNames.length &&
+			variationNamesInDB.every((name) => chosenVariationNames.includes(name));
```

Comment:
That's why i use the key ?

Reply:
but `variationLabels` is only for UI, you cannot rely on it, it could become inconsistent. 

The source of truth is `variations`, we should use it, and not the derived `variationLabels`